src/typed-method-types/workflows/triggers/event.ts

```diff
@@ -1,23 +1,94 @@
 import { BaseTrigger, RequiredInputs, TriggerTypes } from "./mod.ts";
+import { FilterType } from "./trigger-filter.ts";
+import { ObjectValueUnion } from "../../../type-helpers.ts";
 
-type BaseEvent = {
+export const MetadataEventTypes = {
+  MessageMetadataPosted: "slack#/events/message_metadata_posted", //channel
+  // MessageMetadataAdded: "slack#/events/message_metadata_added",
+  // MessageMetadataDeleted: "slack#/events/message_metadata_deleted",
+} as const;
+
+export const TriggerEventTypes = {
+  ChannelCreated: "slack#/events/channel_created", //workspace
+  DndUpdated: "slack#/events/dnd_updated", //workspace
+  ReactionAdded: "slack#/events/reaction_added", //channel
+  UserJoinedChannel: "slack#/events/user_joined_channel", //channel
+  UserJoinedTeam: "slack#/events/user_joined_team", //workspace
+  ...MetadataEventTypes,
+} as const;
+
+//Channel level events
+export const ChannelEventTypes = {
+  MessageMetadataPosted: "slack#/events/message_metadata_posted", //channel
+  ReactionAdded: "slack#/events/reaction_added", //channel
+  UserJoinedChannel: "slack#/events/user_joined_channel", //channel
+} as const;
+
+//Workspace level events
+export const WorkspaceEventTypes = {
+  MessageMetadataPosted: "slack#/events/message_metadata_posted", //channel
+  ChannelCreated: "slack#/events/channel_created", //workspace
+  DndUpdated: "slack#/events/dnd_updated", //workspace
+  UserJoinedTeam: "slack#/events/user_joined_team", //workspace
+} as const;
```

Comment:
Rather than separate objects that define the same values, let's use one object that contains all event types and then use `Pick` for new objects

```ts
type MessageMetadataTypes = Pick <typeof TriggerEventTypes, "MessageMetadataPosted" /* | "MessageMetadataAdded" | "MessageMetadataDeleted" */>
type ChannelTypes = Pick<typeof TriggerEventTypes, "ReactionAdded" | "UserJoinedChannel" | "MessageMetadataPosted">
type WorkspaceTypes = Pick<typeof TriggerEventTypes, "UserJoinedTeam" | "ChannelCreated" | "DndUpdated">
```

Note that this also relies on removing the `MetadataEventTypes` object into the `TriggerEventTypes` and then reconstructing from `Pick`